### ISSUE TITLE
vitest watch mode crash with EDR

### ISSUE THREAD

**Issue by wighawag** [NomicFoundation/edr#385]
### Version of Hardhat

2.22.2

### What happened?

I use 
```
vitest --maxWorkers 1 --minWorkers 1
```

This run a simple test with fixture and whenever I trigger a re launch of the test by modyfing the contract code or the tests, vitest crash with

```
thread '<unnamed>' panicked at crates/edr_napi/src/context.rs:67:14:                                                                                                                                                                      │
│Could not set global default tracing subscriber: SetGlobalDefaultError("a global default trace dispatcher has already been set")                                                                                                          │
│note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace                                                                                                                                                             │
│fatal runtime error: failed to initiate panic, error 5                                                                                                                                                                                    │
│Aborted (core dumped) 
```

### Minimal reproduction steps

```
git clone https://github.com/bug-reproduction/vitest-edr
cd vitest-edr
pnpm i
pnpm test
# modify either solidity or js test and see crash
```

### Search terms

vitest crash

**Comment by Wodann** [comment]
Thank you for the detailed issue description and reproduction steps!

I was able to reproduce the issue and created a fix. We'll release the fix in the next patch release of EDR

**Comment by fvictorio** [comment]
Hi, we just published version 0.3.7 of `@nomicfoundation/edr`, which should fix this.

You can get this version by removing your lockfile and re-installing the packages; this will use the latest EDR version when installing Hardhat. Alternatively, you can wait until the next version of Hardhat, which [will have this bumped dependency](https://github.com/NomicFoundation/hardhat/pull/5173).
